crates/rmf_site_editor/src/widgets/view_multi_selection.rs

```diff
@@ -0,0 +1,119 @@
+use rmf_site_egui::*;
+use rmf_site_format::{InstanceMarker, SiteID};
+use rmf_site_picking::{Select, Selection};
+
+const INSTANCES_VIEWER_HEIGHT: f32 = 200.0;
+
+/// Add a plugin for viewing and editing a list of all levels
+#[derive(Default)]
+pub struct ViewMultiSelectionPlugin {}
+
+impl Plugin for ViewMultiSelectionPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_plugins(PropertiesTilePlugin::<ViewMultiSelection>::new());
+    }
+}
+
+#[derive(SystemParam)]
+pub struct ViewMultiSelection<'w, 's> {
+    icons: Res<'w, Icons>,
+    model_instances:
+        Query<'w, 's, (Entity, &'static NameInSite, &'static SiteID), With<InstanceMarker>>,
+    selection: Res<'w, Selection>,
+    delete: EventWriter<'w, Delete>,
+    select: EventWriter<'w, Select>,
+    multi_edit_pose_widget: MultiEditPoseWidget<'w, 's>,
+}
+
+impl<'w, 's> WidgetSystem<Tile> for ViewMultiSelection<'w, 's> {
+    fn show(_: Tile, ui: &mut Ui, state: &mut SystemState<Self>, world: &mut World) -> () {
+        let mut params = state.get_mut(world);
+        if params.selection.selected.len() < 2 {
+            return;
+        }
+
+        CollapsingHeader::new("ViewMultiSelection")
+            .default_open(true)
+            .show(ui, |ui| {
+                params.show_widget(ui);
+            });
+    }
+}
+
+impl<'w, 's> ViewMultiSelection<'w, 's> {
+    pub fn show_widget(&mut self, ui: &mut Ui) {
+        ScrollArea::vertical()
+            .max_height(INSTANCES_VIEWER_HEIGHT)
+            .show(ui, |ui| {
+                if self.selection.selected.is_empty() {
+                    ui.label("Nothing selected or multiple models selected!");
+                    return;
+                }
+
+                for instance in self.selection.selected.iter() {
+                    let Ok((instance_entity, instance_name, site_id)) =
+                        self.model_instances.get(*instance)
+                    else {
+                        continue;
+                    };
+
+                    ui.horizontal(|ui| {
+                        // Deselect instance from current selection
+                        if ui
+                            .add(Button::image_and_text(
+                                self.icons.deselect.egui(),
+                                format!("#{}", site_id.0),
+                            ))
+                            .on_hover_text("Deselect instance from current selections")
+                            .clicked()
+                        {
+                            self.select.write(Select::new(Some(instance_entity), true));
+                        }
+                        // Delete instance from this site (all scenarios)
+                        if ui
+                            .add(ImageButton::new(self.icons.trash.egui()))
+                            .on_hover_text("Remove instance from all scenarios")
+                            .clicked()
```

Comment:
Notice how the `selector_widget` [also checks for `.hovered()`](https://github.com/open-rmf/rmf_site/blob/b53924b19fbefc6d43ec7d50d66e896f1b35ceea/crates/rmf_site_editor/src/widgets/selector_widget.rs#L64-L66) and then emits the `Hover(_)` event. We should do likewise to help users know which item is being considered for removal before they click.